packages/@eventual/core-runtime/src/providers/executor-provider.ts

```diff
@@ -54,7 +54,7 @@ export class RemoteExecutorProvider<Context = undefined>
     executor: WorkflowExecutor<any, any, any>
   ): Promise<{ storedBytes: number }> {
     // provides a shallow copy of the history events.
-    const historyEvents = executor.history.slice(0);
+    const historyEvents = executor.historyCloned.slice(0);
```

Comment:
maybe just use the cloned one in the executor's iterator and return the original array in `.history`?